src/loaders/ObjectLoader.js

```diff
@@ -323,44 +316,16 @@ class ObjectLoader extends Loader {
 			const loader = new MaterialLoader();
 			loader.setTextures( textures );
 
-			for ( let i = 0, l = json.length; i < l; i ++ ) {
-
-				const data = json[ i ];
-
-				if ( data.type === 'MultiMaterial' ) {
-
-					// Deprecated
-
-					const array = [];
+			for ( const data of json ) {
```

Comment:
Please do not change the structure of for loops.

Reply:
Not sure, why (I think for-of loops are much simpler and easier to read), but OK, reverted.

Reply:
> Not sure, why

Performance